File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/ProfilReferentiel.java
package nesstechnologies.fr.applicationesnbackend.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString(exclude = "referentiel")
public class ProfilReferentiel implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String profil;
    private String niveau;

    @JsonIgnore
    @ManyToOne
    private Referentiel referentiel;

}

File: ats-ness-front-prod/src/app/demo/models/ProfilReferentiel.ts

import {Referentiel} from "./Referentiel";

export class ProfilReferentiel {
    id: number | undefined;

    profil: string | undefined ;
    niveau: string | undefined;

    // Relation avec l'entité Candidat
    referentiel: Referentiel | undefined;
}
